add redshift and bq to supported adapters for freshness (#5080)
dbt-labs/docs.getdbt.com: website/docs/reference/resource-properties/freshness.md

- If a `loaded_at_field` is provided, dbt will calculate freshness via a select query (behavior prior to v1.7).
- If a `loaded_at_field` is _not_ provided, dbt will calculate freshness via warehouse metadata tables when possible (new in v1.7 on supported adapters).

Currently, calculating freshness from warehouse metadata tables is supported on the following adapters:
- [Snowflake](/reference/resource-configs/snowflake-configs)
- [Redshift](/reference/resource-configs/redshift-configs)
- [BigQuery](/reference/resource-configs/bigquery-configs)

> Support is coming soon to the following adapter:

Reviewer: Given this is only a single item being bulleted, we can probably make this a sentence like
```suggestion
Support is coming soon to the [Spark](/reference/resource-configs/spark-configs) adapter.
```

Or something similar

> - [Spark](/reference/resource-configs/spark-configs)

Reviewer: ```suggestion
```

Additionally, the `loaded_at_field` is required to calculate freshness for a table. If a `loaded_at_field` is not provided, then dbt will not calculate freshness for the table.

Freshness blocks are applied hierarchically:

> - a `freshness` and `loaded_at_field` property added to a source will be applied to all tables defined in that source

Reviewer: ```suggestion
- A `freshness` and `loaded_at_field` property added to a source will be applied to all tables defined in that source
```

> - a `freshness` and `loaded_at_field` property added to a source _table_ will override any properties applied to the source.

Reviewer: ```suggestion
- A `freshness` and `loaded_at_field` property added to a source _table_ will override any properties applied to the source.
```
